lib/mongoid/association/embedded/batchable.rb

```diff
@@ -246,6 +246,21 @@ def path
           end
         end
 
+        # Retrieve the path and invalidate the cache afterwards. This method
+        # is used for when the association has not been set on the document
+        # yet, which can cause path and atomic_paths to be calculated
+        # incorrectly later.
+        #
+        # @api private
+        #
+        # @return [ String ] The atomic path.
+        def path_without_cache
```

Comment:
I think this is a confusing method, can it be changed to just clear the cache and the caller look like

```
            _base.delayed_atomic_unsets.delete(path)
￼            clear_atomic_path_cache
￼```

?

Reply:
done